docs/react/README.md

```diff
@@ -1,72 +1,186 @@
 # @hatchifyjs/react
 
-- [What is @hatchifyjs/react](#what-is-hatchifyjsreact)
-  - [TypeScript](#typescript)
-  - [Types](#types)
+@hatchifyjs/react is a schema-driven library of components and helper methods for a Hatchify + React frontend. By providing the schemas (AKA models) of your backend resources, @hatchifyjs/react returns a set of components and helper functionality that you can use across your React app.
+
+`@hatchifyjs/react` is a TypeScript supported [NPM package](https://www.npmjs.com/package/@hatchifyjs/react) that takes your [JSON:API rest client](#createjsonapiclient) and[Schema(s)](../schema/README.md) to produce custom:
+
+- components
+- helper functions
+- generated [types](#types)
+
+The following uses `hatchifyReact`to create a `hatchedReact` app with the defined `Todo` and `User` schemas.
+
+```tsx
+import { hatchifyReact, HatchifyProvider, createJsonapiClient } from "@hatchifyjs/react"
+
+// Define your Schema(s)
+export const Schemas = {
+  Todo: {
+    name: "Todo",
+    attributes: {
+      name: string(),
+      status: enumerate({ values: ["Pending", "Failed", "Completed"] }),
+    },
+    relationships: {
+      user: belongsTo("User"),
+    },
+  },
+  User: {
+    name: "User",
+    attributes: {
+      name: string(),
+    },
+    relationships: {
+      todos: hasMany(),
+    },
+  },
+} satisfies Record<string, PartialSchema>
+
+// Create your Hatched React App instance
+const hatchedReact = hatchifyReact(createJsonapiClient("/api", Schemas))
+
+// Define variables for your Hatchify schema generated components
+const TodoDataGrid = hatchedReact.components.Todo.DataGrid
+
+// Render your React Functional Component
+const HatchedComponent = () => {
+  return (
+    <HatchifyProvider>
+      <button onClick={onActionClick} style={{ margin: 10 }}>
+        action
+      </button>
+      <TodoDataGrid defaultSort={{ direction: "asc", sortBy: "id" }}></TodoDataGrid>
+    </HatchifyProvider>
+  )
+}
+
+export default App
+```
+
+[!IMPORTANT]
+The `HatchifyProvider` wraps all Hatchify components to manage their internal states. See [HatchifyProvider](#hatchifyprovider) for more information.
```

Comment:
`HatchifyProvider` doesn't manage state, it's for figuring out which component (design system to use (user doesn't really need to know this)) and the default components for different types. i'd word it maybe like this:

You must wrap your `App` in a `HatchifyProvider` so that you may use the components provided by Hatchify.

Reply:
also i just checked, the `HatchifyProvider` that is imported from `@hatchifyjs/react` doesn't provide a way to override defaults lol - only the `HatchifyPresentationProvider` which is not exported, i will make a backlog ticket for this

Reply:
Will we be exposing the `HatchifyPresentationProvider` or just allow the `HatchifyProvider` to override defaults